### PR TITLE
added web3 wrapper for maintaining persistent connection

### LINKED ISSUE #8: Build wallet connection UI
`Edit to view comments`

## Requirement: 

```
As a marketplace user,
I want to be able to choose what type of wallet to connect with (metamask/wallet-connect),
So that I can connect with my specific wallet type
```
```
As a marketplace user,
I want to be able to connect my wallet to the dapp,
So I can use the marketplace functionality
```
```
As a marketplace user,
I want to be able to disconnect my wallet from the dapp,
So I can end my session
```
```
As a marketplace user,
I want to be able to be informed if I am on an unsupported network,
So I can change to the correct network
```
- Ability for the user to connect/disconnect their wallet to the polygon network
- Initially have two wallet options: 
   - Metamask
   - Wallet Connect
- UI informing to set to correct network

## Acceptance criteria:

- User can access the marketplace on dev server and connect their wallet via all the connection options

## Dependency:

#18 

## ETA:

- 2 days

## Links:

- https://github.com/NoahZinsmeister/web3-react
- Message Umair for the answers...

### PR DESCRIPTION
**Pull request by abbeydabiri**
Fixes #8 #53 

## Description

Added web3wrapper tag for maintaining persistent connection, this is needed as the current development environment does not maintain persistent connection

### CODE REVIEW

**Comment by anjorinjnr** [comment]
why is this needed? I removed the web3wrapper from the last PR

is there an issue with the current implementation?

**Comment by abbeydabiri** [comment]
> why is this needed? I removed the web3wrapper from the last PR
> 
> is there an issue with the current implementation?

The issue is that a user needs to sign in anytime the page is refreshed.

**Review by jmulq — CHANGES_REQUESTED** [review]
So was the app not persisting the web3 connection when refreshing the page? It was on my local instance weirdly?

File: src/web3/web3wrapper.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  useEffect(() => {
+    web3WrapperInitialize(
+      activateNetwork,
+      networkActive,
+      networkError,
+      web3Loaded,
+      setWeb3Loaded
+    )
+  }, [activateNetwork, networkActive, networkError, web3Loaded, setWeb3Loaded])
+  if (web3Loaded) {
```
[the remark this answers is not in the document]
Comment (abbeydabiri) [reply src/web3/web3wrapper.ts:47]: without the web3wrapper the web3 connection does not persist when the page is refreshed.

[the remark this answers is not in the document]
Comment (abbeydabiri) [reply src/web3/web3wrapper.ts:47]: yes it can be just return children

**Comment by anjorinjnr** [comment]
@abbeydabiri  please take a look this PR https://github.com/Vault-Hill/vhc-marketplace-app/pull/85

We don't need this web3 wrapper it doesn't actually wrap anything.  The logic is already extracted inside the ConnectWallet. There was bug in the update I made that's causing the connection to go away. It's fixed in that PR.

**Pull request closed without merging**